src/core/text_layout_evaluator.js

```diff
@@ -0,0 +1,133 @@
+/* -*- tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+/* Copyright 2012 Mozilla Foundation 
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. 
+ */
+/* globals assert, ColorSpace, DecodeStream, Dict, Encodings,
+           error, ErrorFont, Font, FONT_IDENTITY_MATRIX, fontCharsToUnicode,
+           FontFlags, ImageKind, info, isArray, isCmd, isDict, isEOF, isName,
+           isNum, isStream, isString, JpegStream, Lexer, Metrics,
+           Name, Parser, Pattern, PDFJS, warn, Util, Promise,
+           createPromiseCapability, QuadTree */
+ 
+'use strict';
+ 
+var TextLayoutEvaluator = (function TextLayoutEvaluatorClosure() {
+  function TextLayoutEvaluator() {
+  }
+ 
+  TextLayoutEvaluator.prototype = {
+    addToQuadTree: function TextLayoutEvaluator_addToQuadTree(
+                    obj, id, styles) {
```

Comment:
Nit: please change the formatting of the above lines to either
```javascript
addToQuadTree:
    function TextLayoutEvaluator_addToQuadTree(obj, id, styles) {
```
or
```javascript
addToQuadTree: function TextLayoutEvaluator_addToQuadTree(obj, id,
                                                          styles) {
```
This also applies below.